### PR TITLE
Home and single video listing

### PR DESCRIPTION
**Pull request by Digambar1411**
Added home and single video listing page
files to check : contexts, components

features : filter videos by category
get selected video on single page 
suggestion videos

Note: Add to playlist, watch Later, & Like functionality are yet to implement

### CODE REVIEW

File: src/frontend/components/categoryBar/categoryBar.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { useCategory} from "../../contexts";
+
+const CategoryBar=()=>{
+
+    const { categories,  state :{category}, dispatch} = useCategory();
+   
+    return(
+        <>
+            <div className="main-header">
+                {/* {categories && categories.map(({categoryName,_id})=><button onClick={()=>getCategoryVideos(categoryName)}key={_id} className="category-btn">{categoryName}</button>)} */}
```
Comment (atharvabhanage02) [inline src/frontend/components/categoryBar/categoryBar.js:11]: You can remove commented code before merging

File: src/frontend/components/smallVideoCard/smallVideoCard.css [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+.video-details{
+    display: flex;
+    flex-direction: column;
+    align-content: space-between;
+    gap:10px;
+}
+
+.title{
+    font-size:15px;
+    height:36px;
```
Comment (atharvabhanage02) [inline src/frontend/components/smallVideoCard/smallVideoCard.css:34]: You can make use of `rem`

File: src/frontend/contexts/category-context.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    useEffect(()=>{
+        const getCategories = async ()=>{
+
+            try{
+            const response = await axios.get("/api/categories");
+            setCategories(response.data.categories);
+            }
+            catch(error){
+                console.log(error)
```
Comment (atharvabhanage02) [inline src/frontend/contexts/category-context.js:22]: You can use `console.error` here

File: src/frontend/pages/video listing/video.js [hunk 4]
```diff
@@ -1,5 +1,99 @@
...
+        }
+        getSingleVideo();
+            
+    },[id])
+
+    useEffect(()=>{
+        const getSuggestionVideos = async()=>{
+            try{
+                const response = await axios.get("/api/videos")
+                setSuggestionVideos(response.data.videos.filter(video=>video._id!==id && video.creator===selectedVideo.creator))
```
Comment (atharvabhanage02) [inline src/frontend/pages/video listing/video.js:32]: Liked the Suggested Videos Feature

**Review by atharvabhanage02 — APPROVED** [review]
- UI looks amazing 
- Code is clean and Well structured 
- Suggested few minor changes ,  else everything looks fine and good to merge

**Comment by Digambar1411** [comment]
> * UI looks amazing
> * Code is clean and Well structured
> * Suggested few minor changes ,  else everything looks fine and good to merge

Thanks for review and suggestions

**Review by ashishkatiyar2404 — APPROVED** [review]
UI looks superb. 
Code is also clean, looks good to merge

**Comment by Digambar1411** [comment]
> UI looks superb. Code is also clean, looks good to merge

Thanks for review

**Comment by Digambar1411** [comment]
###

**Pull request merged**